### PR TITLE
Users/mmikkelson/iisfix

### PR DESCRIPTION
**Pull request by mmikkelson**
added ability to pass IPAddress parm to the Add-SslCert function
Also, in spite of the expected behavior for "netsh http add sslcert", omitting the certstorename parameter results in the issue as raised in issue #140
updated and re-ran tests to validate changes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -34,4 +34,5 @@ Extensions/BitBucket/DownloadArtifactBitbucket/bin/*
 Extensions/BitBucket/DownloadArtifactBitbucket/obj/*
 
 ## Nuget 
-nuget.exe
\ No newline at end of file
+nuget.exe
+Extensions/**/DeploymentSDK
```
Comment (mvvsubbu) [inline .gitignore]: remove this

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.ps1 [hunk 2]
```diff
@@ -208,13 +209,13 @@ function Add-SslCert
     }
     else
     {
-        $showCertCmd = [string]::Format("netsh http show sslcert ipport=0.0.0.0:{0}", $port)
+        $showCertCmd = [string]::Format("netsh http show sslcert ipport={0}:{1}", $ipAddress, $port)
```
Comment (mvvsubbu) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.ps1]: When IpAddress is being sent as "All Unassigned" you need to set it to 0.0.0.0

File: Extensions/IISWebAppDeploy/Tests/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.Tests.ps1 [hunk 3]
```diff
@@ -303,7 +303,7 @@ Describe "Tests for verifying Add-SslCert functionality" {
         Mock Run-Command -Verifiable {return "", "" , "" , "", "HostName:port                : localhost:80", "Certificate Hash             : SampleHash"} -ParameterFilter { $failOnErr -eq $false }
 
         $output = Add-SslCert -port "80" -certhash "SampleHash" -hostname "localhost" -sni "true" -iisVersion "8.0" 4>&1 | Out-String
-
+
```
Comment (mvvsubbu) [inline Extensions/IISWebAppDeploy/Tests/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.Tests.ps1]: Pass IpAddress as well in all the tests

File: Extensions/IISWebAppDeploy/Tests/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.Tests.ps1 [hunk 4]
```diff
@@ -312,13 +312,27 @@ Describe "Tests for verifying Add-SslCert functionality" {
...
     Context "Given ipport for cert exists" {
 
         Mock Run-Command { return } -ParameterFilter { }
-        Mock Run-Command {return "", "" , "" , "", "IP:port                      : 0.0.0.0:80", "Certificate Hash             : samplehash"}  -ParameterFilter { $failOnErr -eq $false }
-
-        $output = Add-SslCert -port "80" -certhash "SampleHash" -hostname "localhost" -sni "true" -iisVersion "7.0" 4>&1 | Out-String
+        Mock Run-Command -Verifiable {return "", "" , "" , "", "IP:port                      : 0.0.0.0:80", "Certificate Hash             : samplehash"}  -ParameterFilter { $failOnErr -eq $false }
 
+        $output = Add-SslCert -ipAddress "0.0.0.0" -port "80" -certhash "SampleHash" -hostname "localhost" -sni "false" -iisVersion "8.0" 4>&1 | Out-String
```
Comment (mvvsubbu) [inline Extensions/IISWebAppDeploy/Tests/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.Tests.ps1]: Update this test case for scenario where user passes ipAddress as "All Unassigned" from task UI

File: Extensions/IISWebAppDeploy/Tests/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.Tests.ps1 [hunk 5]
```diff
@@ -327,15 +341,14 @@ Describe "Tests for verifying Add-SslCert functionality" {
...
-    Context "Given hostnameport for cert does not exists" {
+    Context "Given ipport for cert does not exist"{
+        Mock Run-Command {return} -ParameterFilter{}
+        Mock Run-Command {return "", "" , "" , "", "", ""}  -ParameterFilter { $failOnErr -eq $false }
 
-        Mock Run-Command { return }
-        Mock Run-Command {return "", "" , "" , "", "", ""} -ParameterFilter { $failOnErr -eq $false }
-
-        $output = Add-SslCert -port "80" -certhash "SampleHash" -hostname "localhost" -sni "true" -iisVersion "8.0" 4>&1 | Out-String
+        $output = Add-SslCert -ipAddress "0.0.0.0" -port "80" -certhash "SampleHash" -hostname "localhost" -sni "true" -iisVersion "7.0" 4>&1 | Out-String
```
Comment (mvvsubbu) [inline Extensions/IISWebAppDeploy/Tests/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.Tests.ps1]: This you can update for normal Ip scenario

File: .gitignore [hunk 6]
```diff
@@ -34,4 +34,5 @@ Extensions/BitBucket/DownloadArtifactBitbucket/bin/*
 Extensions/BitBucket/DownloadArtifactBitbucket/obj/*
```
Comment (mvvsubbu) [inline .gitignore]: Update change log with fix details under Unreleased Section

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/DeploymentSDK/InvokeRemoteDeployment.ps1 [hunk 7]
```diff
@@ -0,0 +1,332 @@
+﻿$InitializationScript = {
+    function Invoke-PsOnRemote
+    {
+        param(
+            [string]$machineDnsName,
```
Comment (mvvsubbu) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/DeploymentSDK/InvokeRemoteDeployment.ps1]: why is this file part of the review?

File: .gitignore [hunk 8]
```diff
@@ -34,4 +34,5 @@ Extensions/BitBucket/DownloadArtifactBitbucket/bin/*
...
 
 ## Nuget 
-nuget.exe
\ No newline at end of file
+#nuget.exe
+#Extensions/**/DeploymentSDK
```
Comment (mvvsubbu) [inline .gitignore]: Remove this change

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.ps1 [hunk 9]
```diff
@@ -192,6 +193,11 @@ function Add-SslCert
         return
     }
 
+    if($ipAddress -eq "All Unassigned"){
```
Comment (mvvsubbu) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.ps1]: curly braces in next line and proper indentation plz

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/DeploymentSDK/VisualStudioRemoteDeployer.exe.config [hunk 10]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+    <startup useLegacyV2RuntimeActivationPolicy="true"> 
+        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5" />
```
Comment (mvvsubbu) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/DeploymentSDK/VisualStudioRemoteDeployer.exe.config]: There are lot of files which are included in the PR, remove them

**Comment by mvvsubbu** [comment]
Thanks for contribution

File: .gitignore (hunk 6 above)
Comment (mmikkelson) [inline .gitignore]: addressed in commit [9487e13](https://github.com/mmikkelson/vsts-rm-extensions/commit/9487e13813526333b507035f9027f6eda02e6019)

Comment (mmikkelson) [inline .gitignore]: addressed in commit [9487e13](https://github.com/mmikkelson/vsts-rm-extensions/commit/9487e13813526333b507035f9027f6eda02e6019)

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/DeploymentSDK/InvokeRemoteDeployment.ps1 (hunk 7 above)
Comment (mmikkelson) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/DeploymentSDK/InvokeRemoteDeployment.ps1]: addressed in commit [bae6112](https://github.com/mmikkelson/vsts-rm-extensions/commit/bae6112afe5e49f473b944a6fe99992ccbf0df97)

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/DeploymentSDK/VisualStudioRemoteDeployer.exe.config (hunk 10 above)
Comment (mmikkelson) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/DeploymentSDK/VisualStudioRemoteDeployer.exe.config]: addressed in commit [bae6112](https://github.com/mmikkelson/vsts-rm-extensions/commit/bae6112afe5e49f473b944a6fe99992ccbf0df97)

File: .gitignore (hunk 1 above)
Comment (mmikkelson) [inline .gitignore]: addressed in commit [bae6112](https://github.com/mmikkelson/vsts-rm-extensions/commit/bae6112afe5e49f473b944a6fe99992ccbf0df97)

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.ps1 (hunk 2 above)
Comment (mmikkelson) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.ps1]: addressed in commit [165b25c](https://github.com/mmikkelson/vsts-rm-extensions/commit/165b25c5820b95f0dc803b0b1a948d397a9a8a94)

File: Extensions/IISWebAppDeploy/Tests/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.Tests.ps1 (hunk 3 above)
Comment (mmikkelson) [inline Extensions/IISWebAppDeploy/Tests/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.Tests.ps1]: addressed in commit [9e81d78](https://github.com/mmikkelson/vsts-rm-extensions/commit/9e81d78d3efd94c6234d768a897b2c28bf60080d)

**Comment by mmikkelson** [comment]
I don't have permissions to see the details of the failing check, so i'm not sure what to fix at this point...please let me know.

File: Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.ps1 (hunk 9 above)
Comment (mmikkelson) [inline Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppMgmt/AppCmdOnTargetMachines.ps1]: addressed in commit [c26e546](https://github.com/Microsoft/vsts-rm-extensions/pull/159/commits/c26e5461667d1de2cfcd8d567a4b91b433341fd9)

**Comment by mvvsubbu** [comment]
Thanks for your help in fixing the issue, I will take this PR to closure and address remaining issues if any.

**Comment by mvvsubbu** [comment]
Closing this PR, opened a different PR excluding unwanted changes https://github.com/Microsoft/vsts-rm-extensions/pull/171

**Pull request closed without merging**